jmespath/visitor.py

```diff
@@ -1,13 +1,20 @@
 import operator
 
 from jmespath import functions
+from jmespath.functions import is_listlike
 from jmespath.compat import string_type
 from numbers import Number
 
 
+def is_ndarray(arg):
+    return hasattr(arg, "__array_finalize__")
```

Comment:
this could be moved to `jmespath.functions`

Reply:
What about about making this private to this file? I suspect there may be an issue in putting anything mentioning numpy in the public jmespath.py API

Reply:
Now that I've better understood the full purpose of this MR: yes, I think making it private makes more sense. BTW, wouldn't `__array_interface__` be more appropriate? Just a nitpick.

Reply:
Tried various tests for things like DataArray and decided for moving this to functions.py as:

```python
def is_array(arg):
    return hasattr(arg, "__array__")
```